### ISSUE TITLE
[Ads] Java exceptions are thrown when calling AdParam.Builder class methods

### ISSUE THREAD

**Issue by MantasKriksciunas** [EvilMindDevs/hms-unity-plugin#262]
Hi,

I'm having trouble with some of _**HuaweiMobileServices.Ads**_ namespace classes. There seems to be an `java.lang.NoSuchMethodError` exception thrown at runtime when I execute the following code:
```c#
RequestOptions _options = HwAds.RequestOptions;

_options = _options.ToBuilder()
                        .SetTagForChildProtection(TagForChild.TAG_FOR_CHILD_PROTECTION_FALSE)  
                        .SetTagForUnderAgeOfPromise(UnderAge.PROMISE_FALSE)
                        .SetNonPersonalizedAd(NonPersonalizedAd.ALLOW_ALL)
                        .Build();
```

It seems to happen with every `Builder` class `Set*` function. I've tried updating libraries with the latest version from `master` and clearing the project's cache but it didn't help to solve this. 

Perhaps I'm missing something? I could only find these old issues (https://github.com/EvilMindDevs/hms-sdk-unity/issues/46, https://github.com/EvilMindDevs/hms-unity-plugin/issues/56) for reference

Here are the logs: [logcat.txt](https://github.com/EvilMindDevs/hms-unity-plugin/files/7855085/logcat.txt)

Setup
-
Huawei device setup:

- Huawei Y9 2019 (MRD-LX1)
- EMUI version 9.1.0
- Android version 9 (SDK 28)
- HMS Core Version 6.3.0.311

Unity setup:

- `v2.1.0 - Unity 2019/2020/2021` plugin package
- Unity 2020.3.13f1 version
- `Mono` Scripting Backend

Huawei gradle implementations:

- com.huawei.hms:ads-lite:13.4.49.301
- com.huawei.hms:ads-consent:3.4.49.301

**Comment by sametguzeldev** [comment]
Hello there.
Thank you for the beautiful description!

I will be looking into in and I'll get back to you 👍

**Comment by MantasKriksciunas** [comment]
Sorry, it seems that I referenced the wrong class here. After retesting, like you said, this code passes without issues.

Exceptions are related not to `RequestOptions.Builder` but `AdParam.Builder` class instead. Try executing the following code:

```c#
private void Init()
{
      HwAds.Init();
      isInitialized = true;
      adsKitSettings = HMSAdsKitSettings.Instance.Settings;
      Debug.LogWarning("Start");
      
      AdParam.Builder _builder = new AdParam.Builder();
      AdParam _ad_param;

       _ad_param = _builder
            .SetTagForChildProtection(TagForChild.TAG_FOR_CHILD_PROTECTION_FALSE)
            .SetTagForUnderAgeOfPromise(UnderAge.PROMISE_FALSE)
            .SetNonPersonalizedAd(NonPersonalizedAd.ALLOW_ALL)
            .Build();

       Debug.LogWarning("End");
}
```

Also, I don't have any stripping active or proguard files in the project. I've tested this on a fresh project with the only change being the min SDK version, that's set to 21 instead of the default 19.

**Comment by sametguzeldev** [comment]
Hello there @MantasKriksciunas
You are correct. Our wrapper was not sending the correct method signature to the SDK side so it was not able to find the corresponding method. I've fixed this issue in #263. I will be shipping a new version on Monday afternoon.

**Comment by MantasKriksciunas** [comment]
Hello, I'm glad to hear that you've found the problem. Thank you very much!

**Comment by sametguzeldev** [comment]
This issue should be resolved in the [v2.1.1](https://github.com/EvilMindDevs/hms-unity-plugin/releases/tag/v2.1.1-2019-2020) release.

**Comment by alihan98ersoy** [comment]
Hello,
Can you test again. Issue should be resolved.  sametguzeldev@37dd6de8e0b70e8655b82e07c3054a16e478182d
